runtime/service/src/main/java/org/apache/polaris/service/storage/s3/sign/S3RemoteSigningCatalogHandler.java

```diff
@@ -0,0 +1,198 @@
+import org.apache.polaris.core.config.FeatureConfiguration;
+import org.apache.polaris.core.config.RealmConfig;
+import org.apache.polaris.core.context.CallContext;
+import org.apache.polaris.core.entity.CatalogEntity;
+import org.apache.polaris.core.entity.PolarisEntity;
+import org.apache.polaris.core.entity.PolarisEntityConstants;
+import org.apache.polaris.core.persistence.PolarisResolvedPathWrapper;
+import org.apache.polaris.core.persistence.resolver.ResolutionManifestFactory;
+import org.apache.polaris.core.storage.InMemoryStorageIntegration;
+import org.apache.polaris.core.storage.PolarisStorageActions;
+import org.apache.polaris.core.storage.PolarisStorageConfigurationInfo;
+import org.apache.polaris.core.storage.StorageUtil;
+import org.apache.polaris.service.catalog.common.CatalogHandler;
+import org.apache.polaris.service.catalog.common.CatalogUtils;
+import org.apache.polaris.service.catalog.io.FileIOUtil;
+import org.apache.polaris.service.context.catalog.CallContextCatalogFactory;
+import org.apache.polaris.service.s3.sign.model.PolarisS3SignRequest;
+import org.apache.polaris.service.s3.sign.model.PolarisS3SignResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3RemoteSigningCatalogHandler extends CatalogHandler implements AutoCloseable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(S3RemoteSigningCatalogHandler.class);
+
+  private final CallContextCatalogFactory catalogFactory;
+  private final S3RequestSigner s3RequestSigner;
+
+  private CatalogEntity catalogEntity;
+  private Catalog baseCatalog;
+
+  public S3RemoteSigningCatalogHandler(
+      PolarisDiagnostics diagnostics,
+      CallContext callContext,
+      ResolutionManifestFactory resolutionManifestFactory,
+      CallContextCatalogFactory catalogFactory,
+      PolarisPrincipal polarisPrincipal,
+      String catalogName,
+      PolarisAuthorizer authorizer,
+      S3RequestSigner s3RequestSigner) {
+    super(
+        diagnostics,
+        callContext,
+        resolutionManifestFactory,
+        polarisPrincipal,
+        catalogName,
+        authorizer,
+        // external catalogs are not supported for S3 remote signing
+        null,
+        null);
+    this.catalogFactory = catalogFactory;
+    this.s3RequestSigner = s3RequestSigner;
+  }
+
+  @Override
+  protected void initializeCatalog() {
+    catalogEntity =
+        CatalogEntity.of(resolutionManifest.getResolvedReferenceCatalogEntity().getRawLeafEntity());
+    if (catalogEntity.isExternal()) {
+      throw new ForbiddenException("Cannot use S3 remote signing with federated catalogs.");
+    }
+    baseCatalog =
+        catalogFactory.createCallContextCatalog(callContext, polarisPrincipal, resolutionManifest);
+  }
+
+  public PolarisS3SignResponse signS3Request(
+      PolarisS3SignRequest s3SignRequest, TableIdentifier tableIdentifier) {
+
+    LOGGER.debug("Requesting s3 signing for {}: {}", tableIdentifier, s3SignRequest);
```

Comment:
do we need to explicity log it here or quarkus supports Access log, asking because i don't see such logging in Iceberg Catalog Handler

Reply:
Totally Agree, so is the conclusion not to log this in the debug anymore ?

Reply:
TBH its fine, I just wanted to know whats your preference :)